src/modules/organisation/organisation.service.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  BadRequestException,
+  HttpException,
+  HttpStatus,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import { CreateOrganisationDto } from './dto/create-organisation.dto';
+import { UpdateOrganisationDto } from './dto/update-organisation.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Organisation } from './entities/organisation.entity';
+
+@Injectable()
+export class OrganisationService {
+  constructor(
+    @InjectRepository(Organisation)
+    private organisationRepository: Repository<Organisation>
+  ) {}
+
+  create(createOrganisationDto: CreateOrganisationDto) {
+    return 'This action adds a new organisation';
+  }
+
+  async findAll() {
+    try {
+      const orgs = await this.organisationRepository.find();
+      console.log(orgs);
+
+      return orgs;
+    } catch (error) {
+      console.log(error.message);
+    }
+    throw new Error(`Internal error`);
+  }
+
+  findOne(id: number) {
+    return `This action returns a #${id} organisation`;
+  }
+
+  async update(
+    id: string,
+    updateOrganisationDto: UpdateOrganisationDto
+  ): Promise<{ message: string; org: Organisation }> {
+    try {
+      const org = await this.organisationRepository.findOneBy({ id });
+
+      if (!org) {
+        throw new NotFoundException('Organization not found');
+      }
+
+      await this.organisationRepository.update(id, updateOrganisationDto);
+
+      const updatedOrg = await this.organisationRepository.findOneBy({ id });
+
+      if (!updatedOrg) {
```

Comment:
You actually don't need this check here, as the organization existence has been ascertained above before updating the organization and getting here.

Reply:
If any wrong data is provided not conforming to the updateDto, making the update fail, it would return a bad request error.

Reply:
If any issue occur while updating an exception is thrown, so subsequent lines don't get executed. I hope you get the point

Reply:
yes boss, it was an oversight. fixed